controllers/mlmd_test.go

```diff
@@ -820,3 +821,81 @@ func TestDontDeployEnvoyRouteV2(t *testing.T) {
 func boolPtr(b bool) *bool {
 	return &b
 }
+
+func TestGetEndpointsMLMDV2(t *testing.T) {
+	testNamespace := "testnamespace"
+	testDSPAName := "testdspa"
+	expectedMLMDEnvoyName := "ds-pipeline-metadata-envoy-testdspa"
+	expectedMLMDEnvoyRouteName := "ds-pipeline-md-testdspa"
+
+	// Construct DSPA Spec with MLMD Enabled
+	dspa := &dspav1alpha1.DataSciencePipelinesApplication{
+		Spec: dspav1alpha1.DSPASpec{
+			DSPVersion:  "v2",
+			PodToPodTLS: boolPtr(false),
+			APIServer:   &dspav1alpha1.APIServer{},
+			MLMD: &dspav1alpha1.MLMD{
+				Deploy: true,
+				Envoy: &dspav1alpha1.Envoy{
+					Image:       "someimage",
+					DeployRoute: true,
+				},
+			},
+			Database: &dspav1alpha1.Database{
+				DisableHealthCheck: false,
+				MariaDB: &dspav1alpha1.MariaDB{
+					Deploy: true,
+				},
+			},
+			ObjectStorage: &dspav1alpha1.ObjectStorage{
+				DisableHealthCheck: false,
+				Minio: &dspav1alpha1.Minio{
+					Deploy: false,
+					Image:  "someimage",
+				},
+			},
+		},
+	}
+
+	// Enrich DSPA with name+namespace
+	dspa.Namespace = testNamespace
+	dspa.Name = testDSPAName
+
+	// Create Context, Fake Controller and Params
+	ctx, params, reconciler := CreateNewTestObjects()
+	err := params.ExtractParams(ctx, dspa, reconciler.Client, reconciler.Log)
+	assert.Nil(t, err)
```

Comment:
parroting @hbelmiro ... better to change more of these `assert` calls to `require`. Sorry for not mentioning it before -- just (re)learned about that today.

Reply:
There are 143 occurrences of `assert`. Should I change all of them?

Reply:
no, just the ones you added